app/uk/gov/hmrc/agentclientrelationshipsfrontend/models/forms/journey/AgentFastTrackForm.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentclientrelationshipsfrontend.models.forms.journey
+
+import play.api.data.Forms.*
+import play.api.data.format.Formats.*
+import play.api.data.validation.*
+import play.api.data.{Form, Mapping}
+import uk.gov.hmrc.agentclientrelationshipsfrontend.models.AgentFastTrackFormData
+import uk.gov.hmrc.agentclientrelationshipsfrontend.services.ClientServiceConfigurationService
+
+object AgentFastTrackForm {
+  val lowerCaseText: Mapping[String] = of[String].transform(_.trim.toLowerCase, identity)
+  val normalizedText: Mapping[String] = of[String].transform(_.replaceAll("\\s", ""), identity)
+  val uppercaseNormalizedText: Mapping[String] = normalizedText.transform(_.toUpperCase, identity)
+
+  def validateClientId(clientIdRegs: Set[String]): Constraint[String] = Constraint[String] { clientId =>
+    val isValid = clientIdRegs.exists(x => clientId.matches(x))
+    if (isValid) Valid else Invalid(ValidationError(s"INVALID_CLIENT_ID_RECEIVED:${if (clientId.nonEmpty) clientId else "NOTHING"}"))
+  }
+
+
+  def validateFastTrackForm(clientServiceConfig: ClientServiceConfigurationService): Constraint[AgentFastTrackFormData] =
+    Constraint[AgentFastTrackFormData] { agentFastTrackRequest =>
+      val serviceForClientType = agentFastTrackRequest.clientType
+        .fold(true)
+        (clientType => clientServiceConfig.clientServicesFor(clientType).contains(agentFastTrackRequest.service))
+
+      //TODO - confirm if we are going to use clientIdType at all !!
+      val clientIdMatchRegs = clientServiceConfig.clientDetailForServiceAndClientIdType(
+        agentFastTrackRequest.service, 
+        agentFastTrackRequest.clientIdType)
+        .fold(false)(x => agentFastTrackRequest.clientId.matches(x.regex))
+
+      //TODO - should we have some knowFacts regex validation for service - like we have for clientId
```

Comment:
I believe the intent is to encode the possible knownFacts in config with regexes, similar to clientId. Then once the backend returns the required known fact type we can pull the relevant config out, validate and compare to the KF from the BE.